### DISCUSSION TITLE
Why does zoom reset centering?

### DISCUSSION THREAD

**Discussion by TimothyFran** [Mapsui/Mapsui#2145] [General]
Hi all,
I've got a question about zooming and centering.

Why does the map reset its centering when I perform ZoomToLevel?
This is my code:
```
public MainWindow()
{
    InitializeComponent();
    MyMapControl.Map.Layers.Add(OpenStreetMap.CreateTileLayer());

    Debug.WriteLine($"There are {MyMapControl.Map.Layers[0].Resolutions.Count} zoom levels available");

    // Center map on 10.33333, 44.4444444
    CenterMapOnPosition(MyMapControl.Map, 10.33333, 44.4444444);

    ZoomMapToLevel(MyMapControl.Map, 5);

}

private void CenterMapOnPosition(Map map, double lon, double lat)
{

    // Get coordinates in spherical mercator
    MPoint sphericalMercatorPoint = SphericalMercator.FromLonLat(lon, lat).ToMPoint();

    // Center map on coordinates
    map.Home = n => n.CenterOn(sphericalMercatorPoint);

}

private void ZoomMapToLevel(Map map, int level)
{

    // Zoom map at specified level
    map.Home = n => n.ZoomToLevel(level);

}
```
When I call CenterMapOnPosition() the map is centered correctly. But when I call ZoomMapToLevel(), then map centering goes back to somewhere near africa.

I know I can do
`MyMap.Map.Home = n => n.CenterOnAndZoomTo(mercator, MyMap.Map.Layers[0].Resolutions[18]);`
but I need to have the two actions separated.

Is it a bug or am I doing something wrong?

**Comment by pauldendulk** [comment]
You should not use `map.Home` here. It's purpose is to set the default viewport. So, what you see when the application starts. What happens here is that the ZoomMapToLevel method overrides the value of map.Home previously set in CenterMapOnPosition. 

Instead do something like this:
```csharp
private void ZoomMapToLevel(Map map, int level)
{

    // Zoom map at specified level
    map.Navigator.ZoomToLevel(level);
}
```

If you really want to change the default viewport then you should use something like `CenterOnAndZoomTo` as you mentioned.

**Reply by TimothyFran** [reply]
Hi, thanks for your reply.
I changed code like this, but it is not working. Nothing changes in the map.
```
public MainWindow()
{
    InitializeComponent();
    MyMapControl.Map.Layers.Add(OpenStreetMap.CreateTileLayer());

    Debug.WriteLine($"There are {MyMapControl.Map.Layers[0].Resolutions.Count} zoom levels available");

    // Center map on 10.33333, 44.4444444
    CenterMapOnPosition(MyMapControl.Map, 10.33333, 44.4444444);

    ZoomMapToLevel(MyMapControl.Map, 1);

}

private void CenterMapOnPosition(Map map, double lon, double lat)
{
    // Get coordinates in spherical mercator
    MPoint sphericalMercatorPoint = SphericalMercator.FromLonLat(lon, lat).ToMPoint();

    // Center map on coordinates
    map.Navigator.CenterOn(sphericalMercatorPoint);
}

private void ZoomMapToLevel(Map map, int level)
{
    // Zoom map at specified level
    map.Navigator.ZoomToLevel(level);
}
```
[Link to my project](https://filedn.com/l75coKMbsVGJXJ84utNi3Gj/DIY/MapsuiWinuiSample.zip)

**Reply by pauldendulk** [reply]
Thanks for the zip, but I did not inspect it yet. Perhaps you can check some things:
- You say map.Navigator.ZoomToLevel(level) has no effect, but it is the general way to change the viewport used in many of our samples. What if you use some other Navigator method, still no effect?
- If no, check if the `map` that you are using is the map on the mapControl. If you have assigned a new MapControl.Map the older instance will have no effect.
